### PR TITLE
Upgrades jQuery and Bootstrap

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
Upgrades Bootstrap to 4.1.x and jQuery to 3.3.x.  

A vast majority of this PR involves migrating from Bootstrap 3.x to 4.x.  You can see the numerous changes that project made [here](https://getbootstrap.com/docs/4.0/migration/).  All of our old lists now use css grid to allow mixing of static and flex'd columns.  There's also quite a lot of cleanup in this PR.  In several places the col-* classes that we specified did not add up to 12 columns.  I guess this didn't manifest itself as a problem in Bootstrap 3.x but needed to be fixed in 4.x.  I also removed `awx/ui/client/assets/ansible-bootstrap.min.css` entirely.  We'll now be using the base styling shipped with 4.1.x.  All the Bootstrap overrides that _I_ needed to add are in `awx/ui/client/legacy/styles/bootstrap-custom-overrides.less`.  There are other overrides sprinkled in with the legacy css but consolidating them didn't seem like it would gain us very much.

##### ISSUE TYPE
 - Task

##### COMPONENT NAME
 - UI

link https://github.com/ansible/awx/pull/2241 for the original PR

### CODE REVIEW

**Comment by mabashian** [comment]
The lookup field is too tall when compared to other fields in the form:
screen shot 2019-01-09 at 11 29 19 am
 
Settings looks like it's missing panel class:
screen shot 2019-01-10 at 11 48 23 am

Status icons are misaligned:
screen shot 2019-01-10 at 4 42 12 pm

Workflow node form is borked:
screen shot 2019-01-10 at 5 07 50 pm

Org permissions list header:
screen shot 2019-01-11 at 3 18 17 pm

Smart Inventory host filter lookup:
screen shot 2019-01-14 at 1 52 59 pm

### ^^ All of the above is resolved by https://github.com/ansible/awx/pull/2983/commits/e6e64b934c26043320e27441e54dbbd4d66f0d34

File: awx/ui/client/legacy/styles/bootstrap-custom-overrides.less [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    white-space: nowrap;
+}
+
+.dropdown-item.active,
+.dropdown-item:active,
+.dropdown-menu>li>a:hover,
+.dropdown-menu>li>a:focus {
+    text-decoration: none;
+    color: #262626;
+    background-color: #f5f5f5;
```
Comment (jlmitch5) [inline awx/ui/client/legacy/styles/bootstrap-custom-overrides.less]: hex values could be color vars in this file

File: awx/ui/client/src/credentials/factories/kind-change.factory.js [hunk 2]
```diff
@@ -164,19 +163,6 @@ export default
                 scope.authorize = false;
                 scope.authorize_password = null;
             }
-
-            // Collapse or open help widget based on whether scm value is selected
```
Comment (jlmitch5) [inline awx/ui/client/src/credentials/factories/kind-change.factory.js:168]: not sure how this works now this code is removed.

**Review by jlmitch5 — COMMENTED** [review]
Awesome!  Couple small things I noticed when skimming through the code.

File: awx/ui/client/legacy/styles/bootstrap-custom-overrides.less (hunk 1 above)
Comment (mabashian) [reply awx/ui/client/legacy/styles/bootstrap-custom-overrides.less]: 👍

File: awx/ui/client/src/credentials/factories/kind-change.factory.js (hunk 2 above)
Comment (mabashian) [reply awx/ui/client/src/credentials/factories/kind-change.factory.js:168]: To be honest - we might be able to get rid of this entire factory.  The only place it's used is in the adhoc controller when the user hits the reset button and looking at the code for this factory I'm not sure that it actually does anything useful

**Review by jakemcdermott — APPROVED** [review]

**Pull request merged**